subprojects/docs/src/docs/userguide/migrating_from_maven.adoc

```diff
@@ -0,0 +1,518 @@
+
+The Java plugin for Gradle provides a `processResources` task to do the same thing.
+Here's an example configuration:
+
+.Filtering resources via the `processResources` task
+====
+include::sample[dir="userguide/mavenMigration/resourceFiltering/groovy",files="build.gradle[tags=process-resources]"]
+====
+
+So the left hand side of each colon is the token name and the right hand side is a project property.
+This variable substitution will apply to all your resource files (the ones under `src/main/resources` usually).
+
+Gradle has other powerful ways for property processing.
+You can hook in your own filter via a closure that allows you to process the content line by line, or you can add your own `FilterReader` implementation.
+For more details, see the documentation for the link:{javadocPath}/org/gradle/api/file/ContentFilterable.html[ContentFilterable] interface which all copy and archive tasks implement.
+
+
+== Configuring integration tests
+
+Although unit tests are very useful, they can't ensure that an application or library works as a whole.
+It's easy for bugs to appear in the interactions between objects and their interactions with the environment.
+That's why many projects incorporate some form of higher level testing, sometimes termed integration, functional or acceptance testing.
+
+Maven supports these types of test by providing an extra set of phases: `pre-integration-test`, `integration-test`, `post-integration-test`, and `verify`.
+It also uses the Failsafe plugin rather than Surefire so that failed integration tests don't automatically fail the build (because you may need to clean up resources, such as a running application server).
+
+Another factor to consider is where you keep your integration test classes.
+The default approach is to mix them with your unit test classes, but this is less than ideal.
+A common alternative is to use profiles so that you can keep the two types of test separate.
+
+So how should you approach migrating such a setup to Gradle? Forget plugins: source sets are your friends in this situation.
+A standard Java project already has two source sets for your main classes and your unit tests.
+Why not add an extra one for integration tests? Or even more than one for different types of integration test? Say low-level tests against a live database and higher level tests with something like FitNesse.
+
+By declaring a new source set, Gradle automatically sets you up with corresponding configurations (`[sourceSet]Compile` and `[sourceSet]Runtime`) as well as compilation tasks (`compile[SourceSet][Lang]`) and a resource processing task (`process[SourceSet]Resources`).
+All you need to do is add a task to run the tests and ensure that the classpaths are all set up.
+You might also want to add tasks for starting/stopping a database or application server if your tests require something like that.
+
+Let's now take a look at an example so you can see what's involved in practice:
```

Comment:
:o: The following stuff is nearly the very same in the chapter on Testing JVM. Can we trim out most of this subsection and link there? I recognize that maybe there's a bit of specific advice for former Maven users so it might be either tricky and not worth the effort.

Reply:
That chapter didn't exist (or wasn't very good) when this guide was first written. I absolutely agree that this section should link to that chapter. There's no need to reproduce the content. I'll add any necessary caveats or differences the reader needs to be aware of.

Reply:
Done